satellite/console/consoleweb/server.go

```diff
@@ -192,7 +212,282 @@ func (server *Server) appHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-// bucketUsageReportHandler generate bucket usage report page for project
+// authMiddlewareHandler performs initial authorization before every request.
+func (server *Server) authMiddlewareHandler(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		var err error
+		defer mon.Task()(&ctx)(&err)
+		token := getToken(r)
+
+		ctx = auth.WithAPIKey(ctx, []byte(token))
+		auth, err := server.service.Authorize(ctx)
+		if err != nil {
+			ctx = console.WithAuthFailure(ctx, err)
+		} else {
+			ctx = console.WithAuth(ctx, auth)
+		}
+
+		handler.ServeHTTP(w, r.Clone(ctx))
+	})
+}
+
+// tokenRequestHandler authenticates User by credentials and returns auth token.
+func (server *Server) tokenRequestHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	var err error
+	defer mon.Task()(&ctx)(&err)
+
+	type tokenRequestModel struct {
+		Email    string `json:"email"`
+		Password string `json:"password"`
+	}
+
+	var tokenRequest tokenRequestModel
+	err = json.NewDecoder(r.Body).Decode(&tokenRequest)
+	if err != nil {
+		server.serveJSONError(w, 400, err)
+	}
+
+	token, err := server.service.Token(ctx, tokenRequest.Email, tokenRequest.Password)
+	if err != nil {
+		server.serveJSONError(w, 404, err)
+		return
+	}
+
+	err = json.NewEncoder(w).Encode(token)
+	if err != nil {
+		server.log.Error(Error.Wrap(err).Error())
+	}
+}
+
+// changeAccountPasswordRequestHandler updates password for a given user.
+func (server *Server) changeAccountPasswordRequestHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	var err error
+	defer mon.Task()(&ctx)(&err)
+
+	type changePasswordRequestModel struct {
+		Password    string `json:"password"`
+		NewPassword string `json:"newPassword"`
+	}
+
+	var passwordChange changePasswordRequestModel
+	err = json.NewDecoder(r.Body).Decode(&passwordChange)
+	if err != nil {
+		server.serveJSONError(w, 400, err)
+		return
+	}
+
+	err = server.service.ChangePassword(ctx, passwordChange.Password, passwordChange.NewPassword)
+	if err != nil {
+		server.serveJSONError(w, 404, err)
+		return
+	}
+
+	w.WriteHeader(200)
```

Comment:
No need to write `http.StatusOK` it's the default behavior.

Reply:
I thought that here is needed because there is any call to `w.Write`.

On the other hand, the status code should be `204`.